static-instrumenter/maven-plugin/src/main/java/io/opentelemetry/contrib/staticinstrumenter/plugin/maven/PackagingSupportFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter.plugin.maven;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+public final class PackagingSupportFactory {
+
+  private static final Map<String, Supplier<PackagingSupport>> SUPPORTED_FRAMEWORKS =
+      Map.of(
+          "BOOT-INF/lib/", () -> new PackagingSupport("BOOT-INF/classes/"),
+          "WEB-INF/lib/", () -> new PackagingSupport("WEB-INF/classes/"));
+
+  private PackagingSupportFactory() {}
+
+  public static PackagingSupport packagingSupportFor(Path mainArtifact) throws IOException {
+    JarFile jarFile = new JarFile(mainArtifact.toFile());
+    Enumeration<JarEntry> enums = jarFile.entries();
+    while (enums.hasMoreElements()) {
```

Comment:
Wouldn't it be cheaper for iterate over keyset of `SUPPORTED_FRAMEWORKS` and use `getEntry()` to check if a directory entry with the expected name exists?

Reply:
Makes sense.